frontend/src/app/modules/bookings/bookings-page/bookings-page.component.ts

```diff
@@ -42,11 +64,72 @@ export class BookingsPageComponent extends BaseComponent implements OnInit {
             );
     }
 
-    public goToPage(pageName: string) {
+    @HostListener('window:resize', ['$event'])
+    onResize(event: Event) {
+        const eventWindowTarget = event.target as Window;
+
+        const width = eventWindowTarget.innerWidth;
+
+        this.getNumberOfItemsToDisplayOnResize(width);
+    }
+
+    goToPage(pageName: string) {
         this.router.navigate([`${pageName}`]);
     }
 
     followMeetingLink(link: string) {
         window.open(link);
     }
+
+    private getNumberOfItemsToDisplay() {
+        const containerWidth = this.getPageSize();
+
+        switch (true) {
+            case containerWidth < phoneMaxWidth:
+                this.numberOfMembersToDisplay = 1;
+                break;
+            case containerWidth < widthToContainZeroItemUpperLimit:
+                this.numberOfMembersToDisplay = 0;
+                break;
+            case containerWidth < widthToContainTwoLowerLimit:
+                this.numberOfMembersToDisplay = 1;
+                break;
+            case containerWidth < widthToContainTwoUpperLimit:
+                this.numberOfMembersToDisplay = 2;
+                break;
+            case containerWidth < widthToContainThreeItems:
+                this.numberOfMembersToDisplay = 3;
+                break;
+            default:
+                this.numberOfMembersToDisplay = 4;
+                break;
+        }
+    }
+
+    private getNumberOfItemsToDisplayOnResize(width: number) {
+        if (width < desktopWidthToContainThreeItems) {
+            this.numberOfMembersToDisplay = 3;
+        }
+        if (width > desktopWidthToContainFourItems) {
+            this.numberOfMembersToDisplay = 4;
+        }
+        if (width < desktopWidthToContainTwoItems) {
+            this.numberOfMembersToDisplay = 2;
+        }
+        if (width < tabletMaxWidth) {
+            this.numberOfMembersToDisplay = 1;
+        }
+        if (width < phoneMaxWidth) {
+            this.numberOfMembersToDisplay = 1;
+        }
+        if (width < widthToContainZeroItemUpperLimit) {
+            this.numberOfMembersToDisplay = 0;
+        }
```

Comment:
This method differs from previous having different page sizes and is used on resizing page immediately. So that resize will not be work with method  getNumberOfItemsToDisplay() because of order and getting number of item for request will not be working with method getNumberOfItemsToDisplayOnResize(). 
Also I cannot use "return" in this method because "return" for some reason don't change numberOfItems immediately.

Reply:
Refactor these two methods in one.